libsolidity/lsp/Transport.cpp

```diff
@@ -41,6 +42,12 @@ using namespace std;
 using namespace solidity::lsp;
 
 // {{{ Transport
+ofstream Transport::traceLogFileStream() const
+{
+	lspAssert(m_traceLogFilePath, ErrorCode::InternalError, "");
```

Comment:
Also, if the client sets the file path to empty, this will fail. I don't think you can assert here (or even have a validation against this). You just have to handle the case of not having a file name set.